### PR TITLE
Add scrubbing to cybersource

### PR DESCRIPTION
**Pull request by jnormore**
@andrewpaliga @j-mutter @ntalbott @duff a couple reviews please? thanks :)

### CODE REVIEW

File: test/unit/gateways/authorize_net_test.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# encoding: utf-8
```
Comment (jnormore) [inline test/unit/gateways/authorize_net_test.rb]: was causing an error in the ruby 1.9.3 tests

File: test/remote/gateways/remote_cyber_source_test.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ def setup
 
     @gateway = CyberSourceGateway.new({nexus: "NC"}.merge(fixtures(:cyber_source)))
 
-    @credit_card = credit_card('4111111111111111')
+    @credit_card = credit_card('4111111111111111', verification_value: '321')
```
Comment (jnormore) [inline test/remote/gateways/remote_cyber_source_test.rb]: ``assert_scrubbed`` finds ``123`` from the email if we don't use a different cvv here

File: test/unit/gateways/authorize_net_test.rb (hunk 1 above)
Comment (j-mutter) [inline test/unit/gateways/authorize_net_test.rb]: Not supported anymore? :)

File: test/remote/gateways/remote_cyber_source_test.rb [hunk 3]
```diff
@@ -52,6 +52,36 @@ def setup
...
+    credit_card = network_tokenization_credit_card('4111111111111111',
+      :brand              => 'visa',
+      :eci                => "05",
+      :payment_cryptogram => "EHuWW9PiBkWvqE5juRwDzAUFBAk="
+    )
+
+    transcript = capture_transcript(@gateway) do
+      assert auth = @gateway.authorize(@amount, credit_card, @options)
+      assert_success auth
+      assert_equal 'Successful transaction', auth.message
```
Comment (j-mutter) [inline test/remote/gateways/remote_cyber_source_test.rb]: Any reason to assert these here and not in the test above?

**Comment by ntalbott** [comment]
Not sure if it's helpful, but we've used this code in Spreedly for a while:

```ruby
  def scrub(transcript)
    transcript.
      gsub(%r((<wsse:Password [^>]*>)[^<]*(</wsse:Password>))i, '\1[FILTERED]\2').
      gsub(%r((<accountNumber>)[^<]*(</accountNumber>))i, '\1[FILTERED]\2').
      gsub(%r((<cvNumber>)[^<]*(</cvNumber>))i, '\1[FILTERED]\2').
      gsub(%r((<cavv>)[^<]*(</cavv>))i, '\1[FILTERED]\2').
      gsub(%r((<xid>)[^<]*(</xid>))i, '\1[FILTERED]\2').
      gsub(%r((<authenticationData>)[^<]*(</authenticationData>))i, '\1[FILTERED]\2')
  end
```

**Comment by jnormore** [comment]
thanks @ntalbott, I've replace with your version, I'd rather use something you've been using for a while :)

**Comment by j-mutter** [comment]
@ntalbott @jnormore  cool - I was just about to ask if the username/merchant ID should be filtered out. I didn't think they would be considered sensitive, and could be useful when debugging.

File: test/remote/gateways/remote_cyber_source_test.rb (hunk 3 above)
Comment (jnormore) [inline test/remote/gateways/remote_cyber_source_test.rb]: nope, removed :)

**Comment by j-mutter** [comment]
:hatched_chick: once :green_heart:

File: test/unit/gateways/authorize_net_test.rb (hunk 1 above)
Comment (jnormore) [inline test/unit/gateways/authorize_net_test.rb]: riiiigghhttt, removed :)

**Comment by andrewpaliga** [comment]
:+1:

**Pull request closed without merging**